event_record.rb

```diff
@@ -0,0 +1,161 @@
+require 'date'
+require_relative 'event_details'
+class EventRecord
+  @@events_date = {}
+  @@events_details = {}
+  @@id = 0
+  def add_event_details(name, date)
+    begin
+      date = date.strip.split("-")
+      date = DateTime.new(date[0].to_i, date[1].to_i, date[2].to_i)
+      @@id += 1
+      @@events_details[@@id] = Event_Details.new(@@id, name, date)
+      key = date.year.to_s + "-" + date.month.to_s
+      day = date.day
+      if @@events_date.has_key? (key)
+        @@events_date[key].push([name, day])
+      else
+        @@events_date[key] = Array.new
+        @@events_date[key].push([name, day])
+      end
+      print "Event added\n"
+      rescue
+        puts "Incorrect Data"
+    end
+  end
+  
+  def self.return_all_events
+    if @@events_details.length == 0
+      puts "No event added yet"
+    else
+      @@events_details.each do |key, event_detail|
+        puts "ID: #{event_detail.id} Name: #{event_detail.name} Date: #{event_detail.date.year}-#{event_detail.date.month}-#{event_detail.date.day} "
```

Comment:
can you look into date methods from printing

Reply:
done